sdk/storage/azure-storage-blob/src/main/java/com/azure/storage/blob/models/BlobCopyInfo.java

```diff
@@ -0,0 +1,109 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.storage.blob.models;
+
+import com.azure.core.annotation.Immutable;
+import com.azure.storage.blob.BlobAsyncClient;
+import com.azure.storage.blob.BlobClient;
+
+import java.net.URL;
+import java.time.OffsetDateTime;
+import java.util.Objects;
+
+/**
+ * The status of a long running Azure Blob copy operation.
+ *
+ * @see BlobAsyncClient#beginCopyFromUrl(URL)
+ * @see BlobClient#beginCopyFromUrl(URL)
+ */
+@Immutable
+public class BlobCopyInfo {
+    private final String copyId;
+    private final String copySource;
+    private final CopyStatusType copyStatus;
+    private final String eTag;
+    private final OffsetDateTime lastModified;
+    private final String error;
```

Comment:
Is error needed here? Storage will throw error to StorageException. It is not needed for hand craft model.

Reply:
Ah yes. Its not an actual exception. The service call will succeed. The get blob info call returns a set of headers, and it can contain a copy error description, if the copy failed.

Reply:
In case of failure, we get StorageException before you construct this class. Then async emitted a Mono.error()
I did not see any use of the error.

Reply:
If we started a copy operation, and 50% of the way it fails, then it'll set this status description. When we poll, getFileProperties() will return successfully, but have this header information populated. There won't be a StorageException thrown.

https://docs.microsoft.com/en-us/rest/api/storageservices/get-file-properties#response

Reply:
Do you mean at the time of failure, we have the chance of getting status code of 2xx?
Then why don't you construct the class use `headers.getCopyStatusDescription` instead of `headers.getErrorCode()`?

Reply:
No. The service call is "getBlobProperties". The call passes when it is able to get the blob information.

The error is inside the BlobProperties itself, where there is metadata about copy status, if there was something being copied. The REST API explains it.